File: frontend/src/api/sessions.ts
// src/api/sessions.ts
import api from "./client";

export type CreateSessionPayload = {
  domain: string;
  role: string;
  difficulty: string;
  num_questions: number;
};

export type CreateSessionResponse = {
  sessionId: string;
  questions: { id: string; text: string }[];
};

export async function createSession(payload: CreateSessionPayload): Promise<CreateSessionResponse> {
  const res = await api.post("/sessions", payload);
  return res.data as CreateSessionResponse;
}

File: backend/app/api/v1/sessions.py
from fastapi import APIRouter
from pydantic import BaseModel

router = APIRouter()

class SessionCreate(BaseModel):
    domain: str
    role: str
    difficulty: str
    num_questions: int

@router.post('/')
def create_session(data: SessionCreate):
    return {
        'message': 'Session created',
        'data': data
    }
